folly/Traits.h

```diff
@@ -212,6 +212,15 @@ struct like_<Src&&> {
   template <typename Dst>
   using apply = typename like_<Src>::template apply<Dst>&&;
 };
+
+template <template <typename...> class, typename>
+struct is_instantiation_of  : std::false_type {};
```

Comment:
@JoeLoser Could you add some basic comments here? LGTM otherwise as a function in the detail namespace

Reply:
@aary sure, just added some comments describing the trait and its limitations. In the future, if it becomes useful from other parts of the codebase, we can look into adding `is_instantiation_of_value` (or some other name, key part is it can't just be overloaded with `is_instantiation_of` unfortunately) which is just like this except works with values (NTTP) and not types. We still don't have a mechanism to work with the mixing of types and NTTP though as you know.